### PR TITLE
[ticket/14262] Move convertor to controller

### PR DESCRIPTION
**Pull request by CHItA**
[PHPBB3-14262](https://tracker.phpbb.com/browse/PHPBB3-14262)

TODO:
- [x] Add missing language vars
- [x] Connect the controller and converter logic
- [x] Investigate the missing language selector
- [x] Implement navigation stage tracking
- [x] Keep the main Convert menu as active (broken by the use of multiple routes)
- [x] Add missing template structure

### CODE REVIEW

**Comment by marc1706** [comment]
@CHItA bump

**Comment by CHItA** [comment]
basically the convert backend itself is what's missing, i hope it can be copy pasted and then it should be ready for the weekend

**Comment by Nicofuma** [comment]
Bump: status?

**Comment by CHItA** [comment]
https://github.com/phpbb/phpbb/pull/4017/files#diff-5faa55522410c6deccdf20d8942d2c5fR337 and the related legacy file should be connected but i won't have time until wednesday to do anything.

**Comment by Nicofuma** [comment]
ok

**Comment by CHItA** [comment]
@phpbb/development-team ready for testing. it is ugly here and there but works so i don't tell anyone about it if you guys don't

**Comment by CHItA** [comment]
@prototech if you have some time and could have a look, it would be much appreciated

**Comment by Nicofuma** [comment]
Actually it hasn't been converted to the new installer. Just converted to controllers. Rigth?

**Comment by CHItA** [comment]
more or less, some parts been moved but mostly because the corresponding parts of the old installer have been removed

**Comment by Nicofuma** [comment]
In `phpBB/install/convert/convertor.php` you need to implement error and db_error (at least display them in the template please)

Something like 
```
$this->template->assign_block_vars('errors', array(...));
```

File: phpBB/install/convert/convert.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	// Number of rows to be inserted at once (extended insert) if supported
+	// For installations having enough memory a value of 60 may be good.
+	var $num_wait_rows = 20;
+
+	// Mysqls internal recoding engine messing up with our (better) functions? We at least support more encodings than mysql so should use it in favor.
+	var $mysql_convert = false;
+
+	var $p_master;
+
+	function convert(&$p_master)
```
Comment (Nicofuma) [inline phpBB/install/convert/convert.php]: should be

```

	public function __construct($p_master)
	{
		$this->p_master = $p_master;
	}
```

**Comment by Nicofuma** [comment]
besides my 2 comments, :+1:

**Comment by CHItA** [comment]
Done

**Comment by Nicofuma** [comment]
:+1:

File: phpBB/install/convert/controller/convertor.php [hunk 2]
```diff
@@ -0,0 +1,857 @@
...
+		));
+
+		$convertors = $sort = array();
+		$get_info = true; // Global flag
+
+		$handle = @opendir($this->phpbb_root_path . 'install/convertors/');
+
+		if (!$handle)
+		{
+			die('Unable to access the convertors directory');
```
Comment (Elsensee) [inline phpBB/install/convert/controller/convertor.php]: Is it intentional?

Comment (CHItA) [inline phpBB/install/convert/controller/convertor.php]: more or less, that is not the ugliest thing with this anyway

Comment (Elsensee) [inline phpBB/install/convert/controller/convertor.php]: Thought it was a leftover and you wanted to change it into an exception or something like that.

**Pull request merged**